src/managedwriter/stream_connection.ts

```diff
@@ -86,7 +87,22 @@ export class StreamConnection extends EventEmitter {
     this._connection.on('data', this.handleData);
     this._connection.on('error', this.handleError);
     this._connection.on('close', () => {
-      this.trace('connection closed');
+      this.trace('connection closed', this._lastConnectionError);
+      if (this.hasPendingWrites()) {
+        const retrySettings = this._writeClient['_retrySettings'];
+        if (
+          this.isRetryableError(this._lastConnectionError) &&
+          retrySettings.enableWriteRetries
+        ) {
+          this.reconnect();
+          this.resendAllPendingWrites();
+        } else {
+          const err = new Error(
```

Comment:
Consider the [GoogleError class](https://github.com/googleapis/gax-nodejs/blob/80ca2de7cf15c3a547d80b8eccabbcde72929116/gax/src/googleError.ts#L24) from gax possibly?

Reply:
changed to `gax.GoogleError` with status code `ABORTED`